PetGame/Services/PetService.cs

```diff
@@ -200,5 +221,244 @@ public bool DeletePet(ulong petId, ulong ownerId)
+
+            //if this pet doesn't exist, don't go any further. return null
+            if (toReturn == null)
+            {
+                return null;
+            }
+
+            //hunger starts at 100%
+            double hungerPercentage = MaximumPercentage;
+
+            //happiness starts at 100%
+            double happinessPercentage = MaximumPercentage;
+
+            DateTime ToCheck = DateTime.Now.Subtract(TimeSpan.FromDays(NumDaysToCheck));
+            int HoursToCheck = NumDaysToCheck * 24;
+
+            IEnumerable<Activity> PastActivities = activityService.GetActivities(PetId, NumActivities, ToCheck, null);
+
+            //calculate hunger
+            //if there are no activities, decrease hunger by number of hours
+            hungerPercentage = CalculateHunger(PastActivities, hungerPercentage, HoursToCheck);
+
+            //check the number of hours since the last activity
+            int HoursSinceLastActivity = 0;
+            HoursSinceLastActivity = CalculateHoursSinceLastActivity(PastActivities, HoursSinceLastActivity, HoursToCheck);
+
+            int MinutesTillNextAction = 5;
+            MinutesTillNextAction = CalculateMinutesTillNextAction(PastActivities, MinutesTillNextAction, HoursToCheck);
+
+            //multiply HappinessDecrease by the number of hours to get the
+            //happiness percentage
+            happinessPercentage = (HoursSinceLastActivity * (-1 * HappinessDecreasePerHour));
+
+            //ensure that happiness cannot be <0 or >1
+            happinessPercentage = ConstrainPercentages(happinessPercentage);
+
+            //ensure that hunger cannot be <0 or >1
+            hungerPercentage = ConstrainPercentages(hungerPercentage);
+
+            //compile the data into a new PetStatus object
+            //return the new object
+            return new PetStatus() { Pet = toReturn, Hunger = hungerPercentage, Happiness = happinessPercentage,
+                TimeOfNextAction = DateTime.Now.AddMinutes(MinutesTillNextAction)};
+        }//end of function
+
+        private int CalculateHoursSinceLastActivity(IEnumerable<Activity> PastActivities, int HoursSinceLastActivity, int HoursToCheck)
+        {
+            List<Activity> Activities = (List<Activity>)PastActivities;
+            //check to see if there were any activities in the specified interval
+            if (Activities.Count == 0)
+            {
+                HoursSinceLastActivity = HoursToCheck;
+            }
+            else
+            {
+                //get the last activity
+                Activity LastActivity = Activities[0];
+
+                //check the number of hours since the last activity
+                HoursSinceLastActivity = DateTime.Now.Hour - LastActivity.Timestamp.Hour;
+            }
+            return HoursSinceLastActivity;
+        }
+
+        private int CalculateMinutesTillNextAction(IEnumerable<Activity> PastActivities, int MinutesTillNextAction, int HoursToCheck)
```

Comment:
Nitpick, but I do not think that use of `Till` in a name follows good convention.